src/backend/distributed/planner/multi_planner.c

```diff
@@ -296,3 +317,99 @@ CheckNodeIsDumpable(Node *node)
 	pfree(out);
 #endif
 }
+
+
+/*
+ * multi_relation_restriction_hook is a hook called by postgresql standard planner
+ * to notify us about various planning information regarding a relation. We use
+ * it to retrieve restrictions on relations.
+ */
+void
+multi_relation_restriction_hook(PlannerInfo *root, RelOptInfo *relOptInfo, Index index,
+								RangeTblEntry *rte)
+{
+	RelationRestrictionContext *restrictionContext = NULL;
+	RelationRestriction *relationRestriction = NULL;
+	bool distributedTable = false;
+	bool localTable = false;
+
+	if (rte->rtekind != RTE_RELATION)
+	{
+		return;
+	}
+
+	distributedTable = IsDistributedTable(rte->relid);
+	localTable = !distributedTable;
```

Comment:
On 2016-07-22 15:22:10 -0700, Murat Tuncer wrote:
> > +void
> > +multi_relation_restriction_hook(PlannerInfo *root, RelOptInfo *relOptInfo, Index index,
> > +								RangeTblEntry *rte)
> > +{
> > +	RelationRestrictionContext *restrictionContext = NULL;
> > +	RelationRestriction *relationRestriction = NULL;
> > +	bool distributedTable = false;
> > +	bool localTable = false;
> > +
> > +	if (rte->rtekind != RTE_RELATION)
> > +	{
> > +		return;
> > +	}
> > +
> > +	distributedTable = IsDistributedTable(rte->relid);
> > +	localTable = !distributedTable;
> 
> that is to make hasLocalTable, hasDistributedTable assignments in more descriptive way.
> Does ```hasLocalTable |= ! distributedTable``` look better ?

For me these pointless variables makes it noticeably harder to
follow. But it's not important.